07_boolean_2/main.go

```diff
@@ -15,56 +15,262 @@ E.g.:
+	if grade1 >= limit && grade2 >= limit && grade3 >= limit && grade4 >= limit && grade5 >= limit {
+		return true
+	} else {
+		return false
+	}
+}
+
+// check if grade avg >= limit
+func gradeAvgOver(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int, limit int) bool {
+	if (grade1+grade2+grade3+grade4+grade5)/5 >= limit {
+		return true
+	} else {
+		return false
+	}
+}
+
+func calcGrade(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int) string {
+	if gradeValid(grade1, grade2, grade3, grade4, grade5) == false {
+		return "invalid grade"
+	}
+	if gradeAvgOver(grade1, grade2, grade3, grade4, grade5, 90) == true {
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 80) == true {
+			return "A+"
+		} else if allGradesOver(grade1, grade2, grade3, grade4, grade5, 70) == true {
+			return "A"
+		} else if anyGradeUnder(grade1, grade2, grade3, grade4, grade5, 70) == true {
+			return "A-"
+		}
+	}
+	if gradeAvgOver(grade1, grade2, grade3, grade4, grade5, 80) == true {
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 80) == true {
+			return "B+"
+		} else if allGradesOver(grade1, grade2, grade3, grade4, grade5, 70) == true {
+			return "B"
+		} else if anyGradeUnder(grade1, grade2, grade3, grade4, grade5, 60) == true {
+			return "B-"
+		}
+	}
+	if gradeAvgOver(grade1, grade2, grade3, grade4, grade5, 70) == true {
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 70) == true {
+			return "C+"
+		} else if allGradesOver(grade1, grade2, grade3, grade4, grade5, 60) == true {
+			return "C"
+		} else if anyGradeUnder(grade1, grade2, grade3, grade4, grade5, 60) == true {
+			return "C-"
+		}
+	}
+	return "error"
+}
+
+// TODO: Write a function that takes 3 numbers, representing the sides of a
+// triangle, and returns the type of the triangle:
+// - If any of the sides is less than or equal to 0, return "invalid"
+// - If the sum of the lengths of the two shortest sides is less than or
+// equal to the length of the longest side, return "invalid"
+// - If all three sides are equal, return "equilateral"
+// - If two sides are equal, return "isosceles"
+// - If no sides are equal, return "scalene"
+
+// any side <= 0 == true
+func anySideLess(side1 int, side2 int, side3 int) bool {
+	if side1 <= 0 || side2 <= 0 || side3 <= 0 {
+		return true
+	} else {
+		return false
+	}
+}
+
+// sum of length of 2 shortest <= longest side == true
+func sumLength(side1 int, side2 int, side3 int) bool {
+	if side1 <= side2 && side3 > side1 && side3 > side2 || side2 <= side1 && side3 > side1 && side3 > side2 {
+		if side1+side2 <= side3 {
+			return true
+		}
+	} else if side3 <= side1 && side2 > side3 && side2 > side1 || side1 <= side3 && side2 > side3 && side2 > side1 {
+		if side3+side1 <= side2 {
+			return true
+		}
+	} else if side2 <= side3 && side1 > side2 && side1 > side3 || side3 <= side2 && side1 > side2 && side1 > side3 {
+		if side2+side3 <= side1 {
+			return true
+		}
+	}
```

Comment:
Here, consider creating 2 helper functions:

```go
func sumTwoBiggestSides(a int, b int, c int) int {
  // sums the two biggest numbers between a, b, and c
}

func smallestSide(a int, b int, c int) int {
  // returns the smallest value between a, b, and c
}
```

Your function will then be:

```go
func sumLength(side1 int, side2 int, side3 int) bool {
  var sum int = sumTwoBiggestSides(side1, side2, side3)
  var smallest int = smallestSide(side1, side2, side3)

  return sum >= smallest
}
```

Reply:
Why not `sumTwoShortestSides` and `longestSide` according to the requirements?
I did both, but so far it my test for this requirement get me `scalene` instead of `invalid`.

Reply:
Oh yeah, you're right. I got confused!